services/queue/src/api.js

```diff
@@ -192,6 +192,101 @@ builder.declare({
   return res.reply(definition);
 });
 
+/** Get task batched */
+builder.declare({
+  method: 'post',
+  route: '/tasks',
+  query: paginateResults.query,
+  name: 'tasks',
+  scopes: { AllOf: [{
+    for: 'taskId',
+    in: 'taskIds',
+    each: 'queue:get-task:<taskId>',
+  }] },
+  input: 'tasks-request.yml',
+  stability: APIBuilder.stability.experimental,
+  category: 'Tasks',
+  idempotent: true,
+  output: 'tasks-response.yml',
+  title: 'Get multiple task definitions',
+  description: [
+    'This end-point will return the task definition for each input task id.',
+    'Notice that the task definitions may have been modified by queue.',
+    'If an optional property is not specified the queue may provide',
+    'a default value.',
+  ].join('\n'),
+}, async function(req, res) {
+  const taskIds = req.body.taskIds;
+
+  await req.authorize({ taskIds });
+
+  const tasks = [];
+  for (const taskId of taskIds) {
+    // XXX do we bypass the LRUcache and batch query the db above a certain
+    // number of indexes?
+    let task;
+    if (this.LRUcache.has(taskId)) {
+      task = this.LRUcache.get(taskId);
+    } else {
+      task = await Task.get(this.db, taskId);
+      if (task) {
+        this.LRUcache.set(taskId, task);
+      }
+    }
+    // Handle cases where the task doesn't exist
+    if (!task) {
+      return res.reportError('ResourceNotFound', [
+        '`{{taskId}}` does not correspond to a task that exists.',
+        'Are you sure this task has already been submitted?',
+      ].join('\n'), { taskId });
+    }
+    // Create task definition
+    tasks.push({ taskId, task: await task.definition() });
```

Comment:
btw I think this function is not async anymore https://github.com/taskcluster/taskcluster/blob/main/services/queue/src/data.js#L115 
but I see it is used with await in some other places. We can ignore it here and refactor separatly